### PR TITLE
Add support for PRIMARY KEY syntax in column definition

### PR DESCRIPTION
**Pull request by qoega**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Create table now supports `PRIMARY KEY` syntax in column definition. Columns are added to primary index in the same order columns are defined.

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
This is an automated comment for commit 24e77083b38fbfdbec0d5a6fa8da65cb6a33a602 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51881/24e77083b38fbfdbec0d5a6fa8da65cb6a33a602/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Docs Check</td><td>Builds and tests the documentation</td><td>🟡 pending</td></tr>
</table>

**Comment by qoega** [comment]
@DanRoscigno FYI There were many small features missing from the main CREATE TABLE doc page. I added it to syntax for now, but it makes sense to add more cross references to codecs, ephemeral, null/not null.

File: src/Parsers/ASTCreateQuery.h [hunk 1]
```diff
@@ -76,7 +77,7 @@ class ASTColumns : public IAST
         f(reinterpret_cast<void **>(&primary_key));
         f(reinterpret_cast<void **>(&constraints));
         f(reinterpret_cast<void **>(&projections));
-        f(reinterpret_cast<void **>(&primary_key));
+        f(reinterpret_cast<void **>(&primary_key_from_columns));
```
Comment (evillique) [inline src/Parsers/ASTCreateQuery.h:80]:
```suggestion
        f(reinterpret_cast<void **>(&primary_key));
        f(reinterpret_cast<void **>(&primary_key_from_columns));
```

Comment (qoega) [reply src/Parsers/ASTCreateQuery.h:80]: Why? There was a duplicated one already in line 76. I think it was a bug...

**Comment by evillique** [comment]
- ClickHouse Stateless Tests (debug) [5/5] - https://github.com/ClickHouse/ClickHouse/issues/51918

**Review by evillique — APPROVED** [review]

**Pull request merged**